keps/sig-network/4631-loadbalancerstatus/README.md

```diff
@@ -0,0 +1,1210 @@
+    randomly misrouting traffic to nodes that won't accept it.
+
+Currently there is no way for a cloud provider to report failure back
+to the user, and existing implementations vary between "do nothing
+(except log an error)" and "provide a load balancer that won't
+actually work".
+
+We should allow cloud providers to indicate to the user that they
+can't (or won't) provide a load balancer for a given service, and
+require load balancers to do that in cases where they know they can't
+provide at least a minimally-functional load balancer.
+
+```
+<<[UNRESOLVED clouds with multiple load balancer types ]>>
+
+Some cloud providers have multiple types of load balancers, with
+different types having different features. Does it make sense to
+suggest that cloud providers should automatically switch LB types if
+needed to implement the requested functionality? (For example,
+`cloud-provider-aws` currently defaults to "ELB" load balancers, which
+use a `Proxy` style and can't implement source IP preservation. Would
+it make sense for them to automatically switch to "NLB" load balancers
+if the Service is `ExternalTrafficPolicy: Local`?
+
+If so, should we add a field to `LoadBalancerStatus` (or
+`LoadBalancerIngress`) to indicate the LB type that was chosen?
+
+(This question applies to the next section as well.)
+
+<<[/UNRESOLVED]>>
+```
+
+### Expected Behavior When the Cloud Provider Knows It Can't *Fully* Implement a Load Balancer
+
+There are also situations where a cloud provider may be able to
+partially or imperfectly provide load balancing for a Service:
+
+  - The Service uses both TCP and UDP ports, but the cloud/load
+    balancer only supports single-protocol load balancers, so it would
+    only be able to serve one of the two protocols.
+
+  - The Service is dual-stack, but the cloud (or cluster) only
+    supports single-stack load balancers, so it would only be able to
+    serve clients of one IP family.
+
+  - The Service is `ExternalTrafficPolicy: Local` but the cloud cannot
+    implement client source IP preservation, though it *does*
+    implement `HealthCheckNodePort` tracking anyway, so it can still
+    provide "efficient"/"one-hop" load balancing *without* source IP
+    preservation (which might actually be all the user wanted anyway).
+
+  - The Service uses `LoadBalancerSourceRanges` but the cloud does not
+    implement that feature.
+
+  - The Service uses `SessionAffinityConfig` but the cloud does not
+    implement that feature.
+
+(The "mixed TCP and UDP" situation was discussed in [KEP-1435], which
+provided a Service Condition (`LoadBalancerPortsError`) that could be
+used to indicate non-support for this functionality. While the KEP
+required that "All of the major clouds support \[mixed protocols] or
+indicate non-support properly" as a condition for graduating to
+Beta, it seems that only `cloud-provider-gcp` was ever updated to set
+this condition, and that was several months after the feature went
```

Comment:
https://github.com/kubernetes/enhancements/pull/4632/files#r1598418132 and because it was my personal choice, otherwise it will not be implemented ... we should assume not much help here from implementations

Reply:
> we should assume not much help here from implementations

well, the idea is that if they want to run the `[Feature:LoadBalancer]` e2e tests, they'd need to implement whatever we do